contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--4.3.0--4.4.0.sql

```diff
@@ -10566,6 +10566,78 @@ CALL sys.babelfish_alter_default_privilege_on_schema();
 -- Drop this procedure after it gets executed once.
 DROP PROCEDURE sys.babelfish_alter_default_privilege_on_schema();
 
+UPDATE pg_proc p1
+SET probin = (
+    SELECT (
+        jsonb_set(
+            probin::jsonb,
+            '{typmod_array}',
+            to_jsonb(
+                (
+                    SELECT jsonb_agg(
+                        CASE
+                            WHEN a::text = '-8000' AND (p2.proargtypes[b-1] = ('sys.varchar'::regtype)::oid OR p2.proargtypes[b-1] = ('sys.nvarchar'::regtype)::oid OR p2.proargtypes[b-1] = ('sys.varbinary'::regtype)::oid) THEN '-1'
+                            WHEN a::text = '-1' AND p2.proargtypes[b-1] = ('sys.smalldatetime'::regtype)::oid THEN '0'
+                            WHEN a::text = '-1' AND (p2.proargtypes[b-1] = ('sys.varchar'::regtype)::oid OR p2.proargtypes[b-1] = ('sys.nvarchar'::regtype)::oid OR p2.proargtypes[b-1] = ('sys.varbinary'::regtype)::oid OR p2.proargtypes[b-1] = ('sys.nchar'::regtype)::oid OR p2.proargtypes[b-1] = ('sys.binary'::regtype)::oid OR p2.proargtypes[b-1] = ('sys.bpchar'::regtype)::oid) THEN '1'
+                            ELSE a::text
+                        END
+                    )
+                    FROM jsonb_array_elements_text(probin::jsonb->'typmod_array')  WITH ORDINALITY AS elem(a,b)
+                )
+            )
+        )::text
+    )
+)
+FROM (
+    SELECT p.oid, p.proallargtypes, p.proargtypes
+    FROM pg_proc p
+    INNER JOIN sys.babelfish_namespace_ext sch ON sch.nspname = p.pronamespace::regnamespace::name
```

Comment:
Here we aren't updating the typmod for procedure/functions in sys schema. Please make sure that we don't need to update for system objects.

Reply:
I checked system defined functions and procedures defined in PLTSQL language in sys schema. Our changes don't have any affect on them and we don't need to update them.As their typmod value is either a non-default and non-max value or from the datatypes that we are not updating(eg: int4, sysname,text,bit etc).